src/bin/download_sysext.rs

```diff
@@ -281,6 +241,55 @@ fn get_pkgs_to_download<'a>(resp: &'a omaha::Response, glob_set: &GlobSet)
     Ok(to_download)
 }
 
+// Read data from remote URL into File
+async fn fetch_url_to_file<'a, U>(path: &'a Path, input_url: U, client: &'a Client) -> Result<Package<'a>>
+where
+    U: reqwest::IntoUrl + From<U> + std::clone::Clone + std::fmt::Debug,
+    Url: From<U>,
+{
+    let mut file = File::create(path).context(format!("failed to create path ({:?})", path.display()))?;
+
+    let _ = match ue_rs::download_and_hash(client, input_url.clone(), &mut file).await {
+        Ok(ok) => ok,
+        Err(err) => {
+            bail!("unable to download data(url {:?}), err {:?}", input_url, err);
+        }
+    };
+
+    Ok(Package {
+        name: Cow::Borrowed(path.file_name().unwrap().to_str().unwrap()),
```

Comment:
`unwrap` should never be used unless in tests. Always convert to an error with `.ok_or(E)?`

Reply:
done